packages/itwin/quantity-formatting/src/components/quantityformat/FormatSetSelector.tsx

```diff
@@ -0,0 +1,99 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import React from "react";
+import { Flex, Input, List, ListItem, Text } from "@itwin/itwinui-react";
+import { useTranslation } from "../../useTranslation.js";
+
+import type { FormatSet } from "@itwin/ecschema-metadata";
+
+/**
+ * Props for the FormatSetSelector component
+ * @internal
+ */
+interface FormatSetSelectorProps {
+  formatSets?: FormatSet[];
+  selectedFormatSetKey?: string;
+  onFormatSetChange: (formatSet: FormatSet, key: string) => void;
+}
+
+/**
+ * A React component that renders a format set selector with searchable list for choosing format sets.
+ * @beta
+*/
+export const FormatSetSelector: React.FC<FormatSetSelectorProps> = ({
+  formatSets = [],
+  selectedFormatSetKey,
+  onFormatSetChange,
+}) => {
+  const [searchTerm, setSearchTerm] = React.useState("");
+  const { translate } = useTranslation();
+
+  // Filter format sets based on search term
+  const filteredFormatSets = React.useMemo(() => {
+    if (!searchTerm.trim()) {
+      return formatSets;
+    }
+
+    const lowerSearchTerm = searchTerm.toLowerCase();
+    return formatSets.filter((formatSet) =>
+      formatSet.label.toLowerCase().includes(lowerSearchTerm)
+    );
+  }, [formatSets, searchTerm]);
+
+  const handleFormatSetSelect = React.useCallback(
+    (formatSet: FormatSet) => {
+      const key = formatSet.name || `formatSet-${formatSets.indexOf(formatSet)}`;
+      onFormatSetChange(formatSet, key);
+    },
+    [onFormatSetChange, formatSets]
+  );
+
+  const handleSearchChange = React.useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      setSearchTerm(event.target.value);
+    },
+    []
+  );
+
+  return (
+    <Flex flexDirection="column" gap="none" alignItems="flex-start" className="quantityFormat--formatSetSelector-container">
+      <Input
+        value={searchTerm}
+        onChange={handleSearchChange}
```

Comment:
I think this can be inlined `onChange={(e) => setSearchTerm(e.currentTarget.value)}`